### ISSUE TITLE
Automatic dark theme doesn't work again

### ISSUE THREAD

**Issue by mpfaff** [rust-lang/rust#97492]
**I tried this:**

- Set system theme to dark
- Set browser theme to dark or "same as system"
- Make sure doc.rust-lang.org is set to "Use system theme"

**I expected to see this happen:**

[doc.rust-lang.org](https://doc.rust-lang.org/stable/std/index.html) uses the dark theme.

**Instead, this happened:**

[doc.rust-lang.org](https://doc.rust-lang.org/stable/std/index.html) uses the light theme.

**System details:**

Operating system: macOS 12.3.1
Browser: Firefox 100.0.2 **OR** Chromium 102.0.5005.61

**Comment by GuillaumeGomez** [comment]
You say you did on docs.rs:

> * Make sure docs.rs is set to "Use system theme"

And then mention doc.rust-lang.org. So just to be sure: you have done everything you said on only one website, right?

**Comment by mpfaff** [comment]
Apologies, that was supposed to say doc.rust-lang.org. And more specifically, both https://doc.rust-lang.org/stable/std/ and https://doc.rust-lang.org/nightly/std/.

**Comment by GuillaumeGomez** [comment]
So a few possibilities:

 * I'm not sure exactly how web browsers handle system theme change. Did you restart your web browser? (I think you did but just in case...)
 * Can you show us what your local-storage looks like? You can see it in your dev-tools console in the "storage" tab. If you changed the theme setting, it doesn't take into account the environment anymore.

**Comment by mpfaff** [comment]
Sorry for the late response. I have restarted my browser since the issue first cropped up.

This is my local storage in Firefox:
<img width="280" src="https://user-images.githubusercontent.com/23013931/173237329-0ea3fcfc-15b1-42eb-ba82-eee2ebeb3aeb.png">

I found that the tracking protections on Brave were breaking automatic dark mode, so I have gotten the dark mode to work on there now.

As for Firefox, I no longer believe this is a bug in doc.rust-lang.org. After further testing, I have found that Firefox is not exposing the OS dark mode setting to *any* website. Additionally, in Firefox developer tools, the dark mode simulation appears to be completely broken. I have confirmed that the tracking protections in Firefox are not the source of the issue.
